bin/libs/component.sh

```diff
@@ -416,6 +416,153 @@ process_zss_plugin_install() {
+  fi
+}
+
+# $1 = line number
+# $2 = file
+remove_key_value_at_line_number() {
+  mv "$2" "$2.tmp"
+  awk -v n=$1 'NR != n' "$2.tmp" > "$2"
+  rm -f "$2.tmp"
+}
+
+# $1 = key=value
+# $2 = file
+add_key_value_at_end_of_file() {
+  echo "$1" >> "$2"
+}
+
+zis_plugin_install() {
+  plugin_path="${1}"
+  zwes_zis_pluginlib="${2}"
+  zwes_zis_parmlib="${3}"
+  zwes_zis_parmlib_member="${4}"
+  plugin_id="${5}"
+  component_dir="${6}"
+  zwes_zis_parmlib_keys="${7}"
+  parmlib_member_as_unix_file="./${zwes_zis_parmlib_member}.txt"
+
+  if [ ! -f "$parmlib_member_as_unix_file" ]; then
+    copy_mvs_to_uss "${zwes_zis_parmlib}(${zwes_zis_parmlib_member})" "${parmlib_member_as_unix_file}"
+  fi
+
+  changed=0
+  
+  if [ -d "${component_dir}${plugin_path}" ]; then
+    if [ -d "${component_dir}${plugin_path}/loadlib" ] && [ -d "${component_dir}${plugin_path}/samplib" ]; then
+      for module in $(ls ${component_dir}${plugin_path}/loadlib); do # There isn't really a situation where we want to use ZWE_CLI_PARAMETER_ALLOW_OVERWRITE
+        copy_to_data_set "${component_dir}${plugin_path}/loadlib/${module}" "$zwes_zis_pluginlib" "" "true"
+        if [ $? != 0 ]; then
+          exit 1
+        fi
+      done
+      for params in $(ls ${component_dir}${plugin_path}/samplib); do
+        if [ ! -f "${component_dir}${plugin_path}/samplib/${params}" ]; then
+          exit 2
+        fi
+        while read samplib_key_value; do
+          prefix=$(echo "$samplib_key_value" | cut -c -2)
+          if [ "$prefix" = "//" ] || [ "$prefix" = "* " ] || [ "$prefix" = "" ]; then
+            continue
+          fi
+          grep -x "$samplib_key_value" "$parmlib_member_as_unix_file" > /dev/null
+          if [ $? -eq 0 ]; then
+            echo "The key-value pair $samplib_key_value is being skipped because it's already there and hasn't changed."
+            continue
+          else
+            samplib_key=$(get_key_of_string "$samplib_key_value")
+            if [ "$samplib_key" = "" ]; then
+              exit 3
+            fi
+            # In the case of a key not being there, an empty string will be returned.
+            num=$(get_line_number_of_key "$samplib_key" "$parmlib_member_as_unix_file")
+            if [ "$num" != "" ]; then
+              parsed_zwes_zis_parmlib_keys=$(replace "${zwes_zis_parmlib_keys}" "." "_") # replace . with _ in keyname for working key search
+              parsed_samplib_key=$(replace "${samplib_key}" "." "_") # replace . with _ in keyname for working key search
+              samplib_key_value=$(read_json_string ${parsed_zwes_zis_parmlib_keys} ".${parsed_samplib_key}")
+              if [ "$samplib_key_value" = "list" ]; then
+              # The key is comma separated list
+                parmlib_key_value=$(get_string_at_line_number "$num" "$parmlib_member_as_unix_file")
+                parmlib_value=$(get_value_of_string "$parmlib_key_value")
+                samplib_value=$(get_value_of_string "$samplib_key_value")
```

Comment:
Seems like there should be a separate variable to contain whether or not this should be a list, than its value. Looks to me like you are overloading samplib_key_value and the value will end up being "list" in this case, and that's probably not intended.  Either that or it will get an error because it can't find the value of "list", or something.

Reply:
Correct